src/NServiceBus.Testing/Sagas/NonDurableStorage/NonDurableSynchronizedStorageSession.cs

```diff
@@ -3,26 +3,36 @@
     using System;
     using System.Threading;
     using System.Threading.Tasks;
+    using System.Transactions;
+    using Extensibility;
+    using Outbox;
     using Persistence;
+    using Transport;
 
     class NonDurableSynchronizedStorageSession : ICompletableSynchronizedStorageSession
     {
-        public NonDurableSynchronizedStorageSession(NonDurableTransaction transaction)
+        public NonDurableTransaction Transaction { get; private set; }
+
+        public void Dispose()
         {
-            Transaction = transaction;
+            Transaction = null;
         }
 
-        public NonDurableSynchronizedStorageSession()
-            : this(new NonDurableTransaction())
+        public ValueTask<bool> TryOpen(IOutboxTransaction transaction, ContextBag context, CancellationToken cancellationToken = new CancellationToken())
         {
-            ownsTransaction = true;
+            throw new NotImplementedException("Outbox transactions are not supported in the testing framework.");
         }
 
-        public NonDurableTransaction Transaction { get; private set; }
+        public ValueTask<bool> TryOpen(TransportTransaction transportTransaction, ContextBag context, CancellationToken cancellationToken = new CancellationToken())
+        {
+            throw new NotImplementedException("Transport transactions are not supported in the testing framework.");
```

Comment:
Needs to return `false`. Otherwise will blow up in the `LoadHandlersConnector`.

Reply:
As far as I understand, the persistence is only ever used locally here and therefore only needs to cater for the case of open. It will never be called in the LoadHandlersConnector. So technically it doesn't matter right?

Reply:
Correct. There won't be a LoadHandlersConnector, or any pipeline for that matter. Only the Open method is called.